### PR TITLE
LUCENE-10510: Check module access prior to running gjf/spotless tasks

### PR DESCRIPTION
**Pull request by dweiss**

### CODE REVIEW

**Review by uschindler — APPROVED** [review]
Looks fine to me. We can maybe do the same regarding errorprone. Errorprone does not run by default without -Dtests.nightly at moment, but we should do the same.

**Comment by uschindler** [comment]
> Looks fine to me. We can maybe do the same regarding errorprone. Errorprone does not run by default without -Dtests.nightly at moment, but we should do the same.

See here: 

> Looks fine to me. We can maybe do the same regarding errorprone. Errorprone does not run by default without -Dtests.nightly at moment, but we should do the same.

See here: https://github.com/apache/lucene/blob/7a950a3ae6bdb297b97058636ae9f88c2a42e29e/gradle/validation/error-prone.gradle#L20-L26

File: gradle/validation/spotless.gradle [hunk 1]
```diff
@@ -22,6 +22,19 @@
 
 def resources = scriptResources(buildscript)
 
+// Check that GJF has access to jdk.compiler internals (this is set up
+// in gradle JVM options via the generated gradle.properties).
+def checkGoogleFormatHasAccessToJdkInternals = rootProject.tasks.register("checkGoogleFormatHasAccessToJdkInternals", {
+  doFirst {
+    def isExported = ModuleLayer.boot().findModule("jdk.compiler")
+        .get().isExported("com.sun.tools.javac.parser", getClass().module)
```
Comment (uschindler) [inline gradle/validation/spotless.gradle:30]: we should maybe check all required packages?

Comment (dweiss) [reply gradle/validation/spotless.gradle:30]: Yep, we sure can. I'll do this later today.

**Comment by dweiss** [comment]
I've reorganized the build logic a bit to include all expected exported packages and an early errorprone check.

**Review by uschindler — APPROVED** [review]
Perfect! Thanks.

File: gradle/validation/error-prone.gradle [hunk 2]
```diff
@@ -22,7 +22,7 @@ if (rootProject.usesAltJvm && rootProject.runtimeJavaVersion > JavaVersion.VERSI
 }
 
 if (!Boolean.parseBoolean(propertyOrDefault("tests.nightly", "false"))) {
-  skipReason = "skipped on non-nightly runs, pass -Dtests.nightly=true to run"
+  skipReason = "skipped on non-nightly runs, pass -Ptests.nightly=true to run"
```
Comment (uschindler) [inline gradle/validation/error-prone.gradle:25]: Thanks for fixing to use P.

**Pull request merged**